k-frontend/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -220,8 +221,12 @@ public Definition parseDefinitionAndResolveBubbles(
       KompileOptions.MainModule mainModuleName,
       KompileOptions.SyntaxModule mainProgramsModule,
       java.util.Set<Att.Key> excludedModuleTags) {
-    Definition parsedDefinition =
-        parseDefinition(definitionFile, mainModuleName, mainProgramsModule);
+    Definition parsedDefinition;
+    if (options.outerParsedJson) {
+      parsedDefinition = JsonParser.parseDefinition(FileUtil.load(definitionFile));
+    } else {
+      parsedDefinition = parseDefinition(definitionFile, mainModuleName, mainProgramsModule);
+    }
```

Comment:
Maybe pull this out into a function?

Reply:
It might be able to go into `parseDefinition`.